src/payment/strategies/zip/zip-payment-strategy.spec.ts

```diff
@@ -0,0 +1,271 @@
+
+    beforeEach(() => {
+        paymentMethodMock = { ...getZip() };
+        scriptLoader = createScriptLoader();
+
+        store = createCheckoutStore({
+            checkout: getCheckoutState(),
+            customer: getCustomerState(),
+            config: getConfigState(),
+            cart: getCartState(),
+            paymentMethods: getPaymentMethodsState(),
+        });
+
+        jest.spyOn(store, 'dispatch')
+            .mockReturnValue(Promise.resolve(store.getState()));
+
+        jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+            .mockReturnValue(paymentMethodMock);
+
+        jest.spyOn(store, 'getState');
+
+        zipClient = getZipScriptMock('approved');
+
+        zipScriptLoader = new ZipScriptLoader(scriptLoader);
+
+        jest.spyOn(zipScriptLoader, 'load')
+            .mockReturnValue(Promise.resolve(zipClient));
+        jest.spyOn(zipClient.Checkout, 'init');
+
+        requestSender = createRequestSender();
+
+        const paymentClient = createPaymentClient(store);
+        const _requestSender: PaymentMethodRequestSender = new PaymentMethodRequestSender(requestSender);
+        const paymentRequestSender = new PaymentRequestSender(createPaymentClient());
+
+        orderActionCreator = new OrderActionCreator(paymentClient, new CheckoutValidator(new CheckoutRequestSender(createRequestSender())));
+        paymentActionCreator = new PaymentActionCreator(paymentRequestSender, orderActionCreator);
+        paymentMethodActionCreator = new PaymentMethodActionCreator(_requestSender);
+        submitOrderAction = of(createAction(OrderActionType.SubmitOrderRequested));
+        submitPaymentAction = of(createAction(PaymentActionType.SubmitPaymentRequested));
+
+        jest.spyOn(orderActionCreator, 'submitOrder')
+            .mockReturnValue(submitOrderAction);
+        jest.spyOn(paymentActionCreator, 'submitPayment')
+            .mockReturnValue(submitPaymentAction);
+
+        strategy = new ZipPaymentStrategy(
+            store,
+            orderActionCreator,
+            paymentActionCreator,
+            paymentMethodActionCreator,
+            zipScriptLoader
+        );
+
+        jest.spyOn(paymentMethodActionCreator, 'loadPaymentMethod');
+    });
+
+    it('creates an instance of ZipPaymentStrategy', () => {
+        expect(strategy).toBeInstanceOf(ZipPaymentStrategy);
+    });
+
+    describe('#initialize()', () => {
+        let zipOptions: PaymentInitializeOptions;
+
+        beforeEach(() => {
+            zipOptions = { methodId: 'zip' };
+        });
+
+        it('initializes the strategy successfully', async () => {
+            await strategy.initialize(zipOptions);
+
+            expect(paymentMethodActionCreator.loadPaymentMethod).toHaveBeenCalledWith(zipOptions.methodId);
+            expect(zipScriptLoader.load).toHaveBeenCalled();
+        });
+
+        it('does not initialize the strategy if method id is not provided', async () => {
```

Comment:
`methodId` is always defined (as you can see in the TS definition) so you don't really need a test for it.